### PR TITLE
Feature/bclns_brklns_refregs

### LINKED ISSUE #9: Implement `RasGeomHdf.breaklines`
Return breaklines from a HEC-RAS Geometry HDF file as a GeoDataFrame of polylines.

### LINKED ISSUE #10: Implement `RasGeomHdf.bc_lines`
Return boundary condition lines from a HEC-RAS Geometry HDF file as a GeoDataFrame of polylines.

### LINKED ISSUE #11: Implement `RasGeomHdf.refinement_regions`
Return refinement regions from a HEC-RAS Geometry HDF file as a GeoDataFrame of polygons.

### PR DESCRIPTION
**Pull request by bates-j**
This pull request implements the functionality of obtaining boundary condition lines, breaklines, and refinement regions from a 2D RAS geometry HDF file. One unit test per method has been added.

- closes #10 -> 999a7d58cc1cbd718793b5edd2cbfe686d59a399
- closes #9 -> 47dd94c2133bca31ea14ddc8d6cc2600a59595bc
- closes #11 -> f65410bdfcc5ec6131cafe7cbf133d19c7d69b3a

### CODE REVIEW

File: src/rashdf/geom.py [hunk 1]
```diff
@@ -142,13 +142,105 @@ def mesh_cell_faces(self) -> GeoDataFrame:
...
+        v_conv_str = np.vectorize(convert_ras_hdf_string)
+        names = v_conv_str(bc_line_data["Attributes"][()]["Name"])
+        mesh_names = v_conv_str(bc_line_data["Attributes"][()]["SA-2D"])
+        types = v_conv_str(bc_line_data["Attributes"][()]["Type"])
+        multi_lines = list()
+        for pnt_start, pnt_cnt, part_start, part_cnt in bc_line_data["Polyline Info"][()]:
+            points = bc_line_data["Polyline Points"][()][pnt_start:pnt_start+pnt_cnt]
+            parts = bc_line_data["Polyline Parts"][()][part_start:part_start+part_cnt]
+            multi_lines.append(
+                MultiLineString(
```
Comment (thwllms) [inline src/rashdf/geom.py:165]: Can we make this use `LineString` for single-part polylines and `MultiLineString` for multipart? Would be nice to avoid `MultiLineString`s that are actually single-part.

File: src/rashdf/geom.py [hunk 2]
```diff
@@ -142,13 +142,105 @@ def mesh_cell_faces(self) -> GeoDataFrame:
...
+            return GeoDataFrame()
+        bl_line_data = self["/Geometry/2D Flow Area Break Lines"]
+        bl_line_ids = range(bl_line_data["Attributes"][()].shape[0])
+        names = np.vectorize(convert_ras_hdf_string)(bl_line_data["Attributes"][()]["Name"])
+        multi_lines = list()
+        for pnt_start, pnt_cnt, part_start, part_cnt in bl_line_data["Polyline Info"][()]:
+            points = bl_line_data["Polyline Points"][()][pnt_start:pnt_start+pnt_cnt]
+            parts = bl_line_data["Polyline Parts"][()][part_start:part_start+part_cnt]
+            multi_lines.append(
+                MultiLineString(
```
Comment (thwllms) [inline src/rashdf/geom.py:199]: Similar to previous comment -- can we use `LineString` for single-part lines and `MultiLineString` for multi-part?

File: src/rashdf/geom.py [hunk 3]
```diff
@@ -142,13 +142,105 @@ def mesh_cell_faces(self) -> GeoDataFrame:
...
+            return GeoDataFrame()
+        rr_data = self["/Geometry/2D Flow Area Refinement Regions"]
+        rr_ids = range(rr_data["Attributes"][()].shape[0])
+        names = np.vectorize(convert_ras_hdf_string)(rr_data["Attributes"][()]["Name"])
+        multi_polygons = list()
+        for pnt_start, pnt_cnt, part_start, part_cnt in rr_data["Polygon Info"][()]:
+            points = rr_data["Polygon Points"][()][pnt_start:pnt_start+pnt_cnt]
+            parts = rr_data["Polygon Parts"][()][part_start:part_start+part_cnt]
+            multi_polygons.append(
+                MultiLineString(
```
Comment (thwllms) [inline src/rashdf/geom.py:231]: Should this be `Polygon` instead?

**Review by thwllms — CHANGES_REQUESTED** [review]
@bates-j looking good! Just the things we talked about yesterday.

File: src/rashdf/geom.py (hunk 3 above)
Comment (bates-j) [reply src/rashdf/geom.py]: Thanks for catching this! Updated to `Polygon`/`MultiPolygon`.

File: src/rashdf/geom.py (hunk 2 above)
Comment (bates-j) [reply src/rashdf/geom.py]: Updated to use `LineString` when single-part.

File: src/rashdf/geom.py (hunk 1 above)
Comment (bates-j) [reply src/rashdf/geom.py]: Updated to use `LineString` when single-part.

**Review by thwllms — APPROVED** [review]
👌
